File: frontend/event-ticketing-app/src/api/orderService.ts
import apiClient from './axios';
import type { Order, OrderRequest } from '../types';

const orderService = {
  // Create new order
  createOrder: async (orderRequest: OrderRequest): Promise<Order> => {
    const response = await apiClient.post('/orders', orderRequest);
    return response.data.data;
  },

  // Get my orders
  getMyOrders: async (): Promise<Order[]> => {
    const response = await apiClient.get('/orders/my-orders');
    return response.data.data;
  },

  // Get order by ID
  getOrderById: async (id: number): Promise<Order> => {
    const response = await apiClient.get(`/orders/${id}`);
    return response.data.data;
  },

  // Get order by session ID
  getOrderBySessionId: async (sessionId: string): Promise<Order> => {
    const response = await apiClient.get(`/orders/session/${sessionId}`);
    return response.data.data;
  },

  // Confirm order (payment simulation)
  confirmOrder: async (id: number): Promise<Order> => {
    const response = await apiClient.put(`/orders/${id}/confirm`);
    return response.data.data;
  },

  // Cancel order
  cancelOrder: async (id: number): Promise<Order> => {
    const response = await apiClient.put(`/orders/${id}/cancel`);
    return response.data.data;
  },

  // === ADMIN FUNCTIONS ===
  
  // Get all orders (Admin only)
  getAllOrders: async (): Promise<Order[]> => {
    const response = await apiClient.get('/orders');
    return response.data.data;
  },

  // === ORGANIZER FUNCTIONS ===
  
  // Get orders by event ID (Organizer/Admin)
  getOrdersByEventId: async (eventId: number): Promise<Order[]> => {
    const response = await apiClient.get(`/orders/event/${eventId}`);
    return response.data.data;
  },
};

export default orderService;

File: backend/eventticketingplatform/src/main/java/com/nipapager/eventticketingplatform/order/service/OrderService.java
package com.nipapager.eventticketingplatform.order.service;

import com.nipapager.eventticketingplatform.order.dto.OrderDTO;
import com.nipapager.eventticketingplatform.order.request.OrderRequest;
import com.nipapager.eventticketingplatform.response.Response;

import java.util.List;

/**
 * Service interface for order operations
 */
public interface OrderService {

    Response<OrderDTO> createOrder(OrderRequest orderRequest);

    Response<List<OrderDTO>> getMyOrders();

    Response<OrderDTO> getOrderById(Long id);

    Response<OrderDTO> confirmOrder(Long id);

    Response<OrderDTO> cancelOrder(Long id);

    // Admin: Get all orders
    Response<List<OrderDTO>> getAllOrders();

    // Admin: Get orders by event
    Response<List<OrderDTO>> getOrdersByEventId(Long eventId);

    Response<OrderDTO> refundOrder(Long id);
}